src/vs/workbench/contrib/notebook/common/notebookEditorModel.ts

```diff
@@ -169,9 +173,33 @@ export class NotebookEditorModel extends EditorModel implements INotebookEditorM
 		return this;
 	}
 
+	/**
+	 * @description Uses the textmodel resolver service to acquire the untitled file's content
+	 * @param resource The resource that is the untitled file
+	 * @returns The bytes
+	 */
+	private async getUntitledDocumentData(resource: URI): Promise<VSBuffer | undefined> {
+		if (resource.scheme !== Schemas.untitled) {
```

Comment:
Do we want to have this in `EditorModel` so any sub-class can reuse the logic?

Reply:
Right now it's in the Input of the CustomEditor and in the Model of the notebook as I couldn't thread it through if it was in the notebook's input

Reply:
Update, I've moved it to the UntitledTextEditorService and now use that to calculate the value of the untitled file